optuna/samplers/__init__.py

```diff
@@ -44,3 +47,32 @@ def product_search_space(study):
             del search_space[param_name]
 
     return search_space or {}
+
+
+def product_search_space(study):
+    # type: (BaseStudy) -> Dict[str, BaseDistribution]
+    """Return the product search space of the :class:`~optuna.study.BaseStudy`.
+
+    **[NOTICE]**
+    This function is deprecated.
+    Please use :func:`~optuna.samplers.intersection_search_space` instead.
```

Comment:
I found the `deprecated` markup in [this document](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html?highlight=deprecated%3A%3A#directive-deprecated). Do you think if we can use it in the Optuna documentation?